projects/kit/utils/math/slider-key-steps.utils.ts

```diff
@@ -0,0 +1,37 @@
+import {round} from '@taiga-ui/cdk';
+import {TUI_FLOATING_PRECISION} from '@taiga-ui/kit/constants';
+import {TuiKeySteps} from '@taiga-ui/kit/types';
+
+/**
+ * @deprecated DONT USE IT! For internal purposes only. We will remove export of it in next major release.
+ * TODO After deletion deprecations from `InputSlider` (v3.0) move it back to projects/kit/components/slider/slider-key-steps.directive.ts
+ * and remove export of this function
+ *
+ */
+export function findKeyStepsBoundariesByFn(
+    keySteps: TuiKeySteps,
+    fn: ([keyStepPercent, keyStepValue]: [number, number]) => boolean,
+): [[number, number], [number, number]] {
+    const keyStepUpperIndex = keySteps.findIndex((ketStep, i) => i && fn(ketStep));
+    const lowerStep = keySteps[keyStepUpperIndex - 1];
+    const upperStep = keySteps[keyStepUpperIndex];
+
+    return [lowerStep, upperStep];
+}
+
+/**
+ * @deprecated DONT USE IT! For internal purposes only. We will remove export of it in next major release.
+ * TODO After deletion deprecations from `InputSlider` (v3.0) move it back to projects/kit/components/slider/slider-key-steps.directive.ts
+ * and remove export of this function
+ */
+export function transformToControlValue(
+    valuePercentage: number,
+    [upperStepPercent, upperStepValue]: [number, number],
+    [lowerStepPercent, lowerStepValue]: [number, number],
+): number {
+    const ratio =
+        (valuePercentage - lowerStepPercent) / (upperStepPercent - lowerStepPercent);
+    const controlValue = (upperStepValue - lowerStepValue) * ratio + lowerStepValue;
+
+    return round(controlValue, TUI_FLOATING_PRECISION);
+}
```

Comment:
can we write a simple unit test for these functions? so then we can be refactored if needed

Reply:
all math calculation is already tested [here](https://github.com/Tinkoff/taiga-ui/blob/main/projects/kit/components/slider/test/slider-key-steps.spec.ts).

Should we duplicate checks for this functions ?

Reply:
oh, sorry, no